frame/child-bounties/src/lib.rs

```diff
@@ -0,0 +1,892 @@
+	parent_bounty: BountyIndex,
+	/// The child bounty curator fee.
+	fee: Balance,
+	/// The deposit of child-bounty curator.
+	curator_deposit: Balance,
+	/// The status of this child-bounty.
+	status: ChildBountyStatus<AccountId, BlockNumber>,
+}
+
+/// The status of a child-bounty.
+#[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+pub enum ChildBountyStatus<AccountId, BlockNumber> {
+	/// The child-bounty is added and waiting for curator assignment.
+	Added,
+	/// A curator has been proposed by the parent-bounty curator. Waiting for acceptance from the
+	/// child-bounty curator.
+	CuratorProposed {
+		/// The assigned child-bounty curator of this bounty.
+		curator: AccountId,
+	},
+	/// The child-bounty is active and waiting to be awarded.
+	Active {
+		/// The curator of this child-bounty.
+		curator: AccountId,
+	},
+	/// The child-bounty is awarded and waiting to released after a delay.
+	PendingPayout {
+		/// The curator of this child-bounty.
+		curator: AccountId,
+		/// The beneficiary of the child-bounty.
+		beneficiary: AccountId,
+		/// When the child-bounty can be claimed.
+		unlock_at: BlockNumber,
+	},
+}
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	#[pallet::generate_storage_info]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config:
+		frame_system::Config + pallet_treasury::Config + pallet_bounties::Config
+	{
+		/// Maximum number of child-bounties that can be added to a parent bounty.
+		#[pallet::constant]
+		type MaxActiveChildBountyCount: Get<u32>;
+
+		/// Minimum value for a child-bounty.
+		#[pallet::constant]
+		type ChildBountyValueMinimum: Get<BalanceOf<Self>>;
+
+		/// The overarching event type.
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+		/// Weight information for extrinsics in this pallet.
+		type WeightInfo: WeightInfo;
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// The parent bounty is not in active state.
+		ParentBountyNotActive,
+		/// The bounty balance is not enough to add new child-bounty.
+		InsufficientBountyBalance,
+		/// Number of child-bounties exceeds limit `MaxActiveChildBountyCount`.
+		TooManyChildBounties,
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// A child-bounty is added.
+		ChildBountyAdded { index: BountyIndex, child_index: BountyIndex },
+		/// A child-bounty is awarded to a beneficiary.
+		ChildBountyAwarded {
+			index: BountyIndex,
+			child_index: BountyIndex,
+			beneficiary: T::AccountId,
+		},
+		/// A child-bounty is claimed by beneficiary.
+		ChildBountyClaimed {
+			index: BountyIndex,
+			child_index: BountyIndex,
+			payout: BalanceOf<T>,
+			beneficiary: T::AccountId,
+		},
+		/// A child-bounty is cancelled.
+		ChildBountyCanceled { index: BountyIndex, child_index: BountyIndex },
```

Comment:
A block explorer will always show the events prefixed with the pallet name, so it would end up being `ChildBounties::ChildBountyAwarded`, I'd consider removing the prefix. 

```suggestion
	pub enum Event<T: Config> {
		/// A child-bounty is added.
		Added { index: BountyIndex, child_index: BountyIndex },
		/// A child-bounty is awarded to a beneficiary.
		Awarded {
			index: BountyIndex,
			child_index: BountyIndex,
			beneficiary: T::AccountId,
		},
		/// A child-bounty is claimed by beneficiary.
		Claimed {
			index: BountyIndex,
			child_index: BountyIndex,
			payout: BalanceOf<T>,
			beneficiary: T::AccountId,
		},
		/// A child-bounty is cancelled.
		Canceled { index: BountyIndex, child_index: BountyIndex },
```